v2/src/features/UploadList/UploadList.tsx

```diff
@@ -52,11 +52,44 @@ const generateErrorPayload = (error: AxiosError | Error) => {
   return errorPayload;
 };
 
+const LOCAL_STORAGE_KEY = 'sugar-share-files';
+
+const setOnLocalStorage = (key: string, value: any) => {
+  if (chrome.storage) {
+    chrome.storage.local.set({ key: value }, function () {
+      console.log(`Saved ${key} to local storage`, value);
+    });
+  } else {
+    window.localStorage.setItem(key, JSON.stringify(value));
+  }
+};
+
+const getOnlocalStorage = (key: string) => {
+  if (chrome.storage) {
+    chrome.storage.local.get([key], function (result) {
+      console.log(`Retrieved ${key} from local storage`, result);
+      return result;
+    });
+  } else {
+    const result = window.localStorage.getItem(key) || '[]';
+    return JSON.parse(result);
+  }
+};
+
+const INIT_STATE = getOnlocalStorage(LOCAL_STORAGE_KEY)
+  ? (getOnlocalStorage(LOCAL_STORAGE_KEY) as SugarFileState[])
+  : ([] as SugarFileState[]);
```

Comment:
if there is something in the local storage start the state with it, if not just use a empty array